specs/configuration_files_specification.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Configuration Files Specification
+layout: posts
+version: 1
+---
+
+# Configuration Files Specification
+
+## Introduction
+
+Various specifications attempt to define configuration files and file formats. This
+specification establishes where these files should be looked for, in which
+order, and how precedence, masking, extensions and overrides work.
+
+The purpose of the rules defined here is to allow OS vendors to implement the
+hermetic-usr pattern, where all vendor files are shipped in the vendor tree itself
+(`/usr/`), including configuration files with system defaults, while allowing local
+or vendor overrides without modifying the original files, for easier management. This is
+especially beneficial for image-based deployments, where the vendor tree is read-only.
+
+These rules are derived from existing real-world usage from the [systemd
+project](https://github.com/systemd/systemd) and the [libeconf
+project](https://github.com/openSUSE/libeconf).
+
+This specification is agnostic toward the actual format and content of the configuration
+files, the precise path used under each top-level hierarchy, and also toward the
+filenames and extensions, with the exception of the `.d/` notation for drop-ins
+directories.
+
+## Storage Directories and Overrides
+In order to allow shipping system defaults owned by the OS vendor, while at the same
+time letting local users or admins override those defaults, `/usr/` and `/etc/` are both
+supported for storage of configuration files, with the latter having higher priority.
+Programs must work correctly if no configuration files are found in `/etc/`.
+Optionally, `/run/` is also supported for ephemeral overrides.
+
+For example, `/usr/foo/bar.conf` provides the default configuration file, but a user
+can create `/etc/foo/bar.conf` which would take precedence and completely override
+the former. If `/run/foo/bar.conf` is present and supported, it would take precedence
+over `/usr/foo/bar.conf`, but not `/etc/foo/bar.conf`, which would still have the
+highest priority.
+
+## Masking
+As a special override case, it must be possible to mask files across different
+locations by creating a symlink to `/dev/null` or an empty file.
+
+For example, an empty `/etc/foo/bar.conf` means that `/usr/foo/bar.conf` is
+masked and thus not parsed.
+
+## Drop-ins
+All configuration paths must support drop-ins. Supporting drop-ins means that in
+addition to parsing a full configuration file, an implementation also must parse all
+files in the drop-in directory associated with it, and merge the result in lexicographic
+order. Drop-ins are not full replacements, but must allow specifying a subset of settings
+present in the main configuration files.
+
+For example, in addition to parsing `/etc/foo/bar.conf`, `/etc/foo/bar.conf.d/a.conf` and
+`/etc/foo/bar.conf.d/b.conf` must also be parsed, in this order.
```

Comment:
this should make clear somehow that /etc/foo/bar.conf.d/*.conf is loaded, i.e. not just a and b, but *everything*

Reply:
previous paragraph says `an implementation also must parse all files in the drop-in directory` which should be clear, in the example here I also wanted to exemplify the ordering